### PR TITLE
Feature/navigation bar

### PR DESCRIPTION
**Pull request by teaskvarc**

### CODE REVIEW

**Comment by teaskvarc** [comment]
TO DO:
1. Search

File: components/UiKit/LinkAsButton.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import styled from "styled-components";
+
+const LinkAsButton = styled.a`
```
Comment (mitjapotocin) [inline components/UiKit/LinkAsButton.tsx:3]: This is fine for now, but we will probably change this to a generic button component.
If you want to use the styled.button as a `<a>` tag, you can just use it like this: 
`<Button as="a" href="#">Link with Button styles</Button>`

See documentation: https://styled-components.com/docs/api#as-polymorphic-prop

File: components/Navbar/Navbar.tsx [hunk 2]
```diff
@@ -3,47 +3,154 @@ import Link from "next/link";
...
+      
+          <MenuWrapper navOpened={navOpened}>
+            <MenuList>
+              {config.menu.map(({ name, url }) => {
+                return (
+                  <li key={name}>
+                    <Link href={url}>{name}</Link>
+                  </li>
+                );
+              })}
```
Comment (mitjapotocin) [inline components/Navbar/Navbar.tsx:143]: I know I wrote this part, but let's do an implicit return here.

```suggestion
              {config.menu.map(({ name, url }) => (
                <li key={name}>
                  <Link href={url}>{name}</Link>
                </li>
              ))}
```

File: components/Navbar/Navbar.tsx [hunk 3]
```diff
@@ -3,47 +3,154 @@ import Link from "next/link";
...
+import BurgerButton from "./BurgerButton/BurgerButton";
+import Image from "../../components/Image/Image";
+import LogoImage from "../../public/assets/icons/logo-orange.svg";
+import Adapt from "@components/UiKit/Adapt";
+import LinkAsButton from "@components/UiKit/LinkAsButton";
 
-const Nav = styled.nav<{ $open?: boolean }>`
-  background: ${({ theme }) => theme.orangeColor};
-  height: 60px;
+const Nav = styled.nav`
```
Comment (mitjapotocin) [inline components/Navbar/Navbar.tsx:12]: This Navbar commponent is looking a bit crowded now.
I recomend moving styled-components into Navbar.styled.tsx (in the same folder)

File: components/Navbar/BurgerButton/BurgerButton.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+`;
+
+function BurgerButton({ onClick }: { onClick: any }){
+    
+    return (
+        <StyledBurger onClick={onClick}>
+          <div />
+          <div />
+          <div />
+          <span className="sr-only">Toggle navigation</span>
```
Comment (mitjapotocin) [inline components/Navbar/BurgerButton/BurgerButton.tsx:40]: Let's stick to styled components, and don't mix class-based styles.
You can created a `styled.span` named SrOnly and put it in UiKit folder.

File: components/Navbar/BurgerButton/BurgerButton.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    div {
+        width: 22px;
+        height: 2px;
+        background: ${({ theme }) => theme.white};
+        transform-origin: 1px;
+        transition: all 0.3s linear;
+      }
+`;
+
+function BurgerButton({ onClick }: { onClick: any }){
```
Comment (mitjapotocin) [inline components/Navbar/BurgerButton/BurgerButton.tsx:33]: We know the type of `onClick` 

```suggestion
function BurgerButton({ onClick }: { onClick: () => void }) {
```

File: components/Navbar/Navbar.tsx [hunk 6]
```diff
@@ -3,47 +3,154 @@ import Link from "next/link";
...
-      <Burger onClick={() => setOpen(!open)}>X</Burger>
+    <Nav>
+      <Adapt>
+        <NavInner>
+
+          <Link href="/">
+            <Image className="img-logo" src={LogoImage.src} width={LogoImage.width} height={LogoImage.height} alt="Orange Logo"/>
+          </Link>
+      
+          <MenuWrapper navOpened={navOpened}>
```
Comment (mitjapotocin) [inline components/Navbar/Navbar.tsx:135]: I would suggest using a `$` prefix for styled component props, so you can easily distinguish them from HTML attributes.

```suggestion
          <MenuWrapper $navOpened={navOpened}>
```

File: components/Navbar/Navbar.tsx [hunk 7]
```diff
@@ -3,47 +3,154 @@ import Link from "next/link";
...
+  display: flex;
+  flex-wrap: wrap;
+  align-items: center;
+  height: 100%;
+
+  .img-logo {
+    width: 115px;
+    margin-top: 18px;
+  }
+`;
```
Comment (mitjapotocin) [inline components/Navbar/Navbar.tsx:32]: There should be one empty line between every component

**Review by mitjapotocin — CHANGES_REQUESTED** [review]
I left some comments, otherwise, it looks good 👌 Good job.

I noticed you are not using auto-formatting, or you have a different configuration. Tell me if you need help setting it up, because this need to be formatted correctly before merging.

File: components/Navbar/BurgerButton/BurgerButton.tsx [hunk 8]
```diff
@@ -1,45 +1,45 @@
...
-          <span className="sr-only">Toggle navigation</span>
-        </StyledBurger>
-    );
+function BurgerButton({ onClick }: { onClick: () => void }) {
+  return (
+    <StyledBurger onClick={onClick}>
+      <div />
+      <div />
+      <div />
+      <SrOnly>Toggle navigation</SrOnly>
```
Comment (mitjapotocin) [inline components/Navbar/BurgerButton/BurgerButton.tsx:40]: @teaskvarc I thik `aria-label` on the burger would be suficient here. We don't need SrOnly

**Pull request merged**